src/Functions/FunctionsHashing.h

```diff
@@ -245,10 +245,11 @@ struct HalfMD5Impl
             uint64_t uint64_data;
         } buf;
 
-        MD5_CTX ctx;
-        MD5_Init(&ctx);
-        MD5_Update(&ctx, reinterpret_cast<const unsigned char *>(begin), size);
-        MD5_Final(buf.char_data, &ctx);
+        EVP_MD_CTX* ctx = EVP_MD_CTX_new();
+        EVP_DigestInit_ex(ctx, EVP_md5(), nullptr);
```

Comment:
These could theoretically return an error, no? I think it is important to check as any of these hash functions may be disabled in FIPS build.

The same applies to the next file.